platforms/core-runtime/launcher/build.gradle.kts

```diff
@@ -74,21 +74,22 @@ dependencies {
     implementation(libs.commonsLang)
     implementation(libs.ant)
 
-    runtimeOnly(project(":bootstrap"))
+    runtimeOnly(project(":gradle-main"))
     runtimeOnly(project(":declarative-dsl-provider"))
     runtimeOnly(project(":problems"))
 
     runtimeOnly(libs.commonsIo)
     runtimeOnly(libs.commonsLang)
     runtimeOnly(libs.slf4jApi)
 
-    manifestClasspath(project(":bootstrap"))
+    manifestClasspath(project(":gradle-main"))
     manifestClasspath(projects.javaLanguageExtensions)
     manifestClasspath(project(":base-services"))
     manifestClasspath(project(":worker-services"))
     manifestClasspath(project(":core-api"))
     manifestClasspath(project(":core"))
     manifestClasspath(project(":persistent-cache"))
+    manifestClasspath(project(":concurrent"))
```

Comment:
❔ How come this is needed here now? What are the rules of the `manifestClasspath`?

Reply:
It was always needed, we just didn't know it. There's a failure path in the client that uses types from ":concurrent" and this would blow up with a "class not found" exception. There's no test for this because it's the last fallback in a chain of fallbacks, so it only gets hit when we've made some kind of mistake lower down (which I happened to do). Using the runtime classpath of `gradle-main` instead of a hardcoded list would have prevented this problem.